### ISSUE TITLE
Wheel Time Picker

### ISSUE THREAD

**Issue by paul1tsch** [shadcn-ui/ui#5305]
### Feature description

something similar to this:

![Wheel Time Picker (iOS)](https://github.com/user-attachments/assets/a69cde43-91af-4f15-aea6-9e37bb4b51e4)

### Affected component/components

DatePicker

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues and PRs

**Comment by huybuidac** [comment]
@paul1tsch This might be useful.
Demo: https://shadcn-datetime-picker-xi.vercel.app/
Repo: https://github.com/huybuidac/shadcn-datetime-picker/blob/main/components/simple-time-picker.tsx
<img width="302" alt="image" src="https://github.com/user-attachments/assets/17d75825-bca1-48dd-91b1-f92cff8a2c1b">

**Comment by huybuidac** [comment]
@NNBnh Yep, it can pick dates too <3 https://shadcn-datetime-picker-pro.vercel.app/?path=/story/datetimepicker--default

![image](https://github.com/user-attachments/assets/c5faedd1-a54d-49fd-b0c3-1b5bde4e7b51)

**Comment by srini047** [comment]
Pretty much what I was looking. Manifesting a `npx shadcn@latest add datetime-picker`

**Comment by shadcn** [comment]
This issue has been automatically marked as stale due to two years of inactivity. It will be closed in 7 days unless there’s further input. If you believe this issue is still relevant, please leave a comment or provide updated details. Thank you.

**Comment by shadcn** [comment]
No stale. Something went wrong with the stale bot.

**Comment by Aravind-pk** [comment]
I would also like a generic scroll-snap wheel picker component to create time/month pickers like this.

![Image](https://github.com/user-attachments/assets/1154f92b-ec4e-4591-a1f3-d88c4f4f5b9b)

**Comment by YThinker** [comment]
https://www.embla-carousel.com/examples/predefined/#ios-style-picker-default
shadcn/ui uses the Embla Carousel library to build <Carousel /> components/

<img width="1430" alt="Image" src="https://github.com/user-attachments/assets/c03873e0-3981-4448-9bf2-e055aefc3b5f" />

**Comment by sahandsn** [comment]
i created one for date, i think you can easily customize it for time as well.
[https://ios-date-picker-shadcn-ui.vercel.app/](https://ios-date-picker-shadcn-ui.vercel.app/)
